proxy/http3/Http3Session.cc

```diff
@@ -55,6 +55,7 @@ void
 HQSession::remove_transaction(HQTransaction *trans)
 {
   this->_transaction_list.remove(trans);
+  delete trans;
```

Comment:
I feel like I want to delete transactions on the caller side, but I'll propose the change later.

Reply:
This is fine, but I wonder why it leaks. Because all transactions in _transaction_list are deleted in the destructor of HQSession. See #9346

Reply:
If `HQSession::remove_transaction()` is called it will remove the transaction from the `_transaction_list` and it will never be deleted.

Reply:
Ah, you're right. Go ahead and merge this.